cmd/cmd.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"fmt"
+)
+
+// DatastoreEngine type to define different engine types
+type DatastoreEngine string
+
+// Types of Datastore Engines
+const (
+	Memory   DatastoreEngine = "memory"
+	Postgres DatastoreEngine = "postgres"
+	MySQL    DatastoreEngine = "mysql"
+)
+
+// String converts DatastoreEngine e to string
+func (e DatastoreEngine) String() string {
+	return string(e)
+}
+
+// IsValid checks if DatastoreEngine e is valid
+func (e DatastoreEngine) IsValid() bool {
+	for _, engineType := range []DatastoreEngine{Memory, Postgres, MySQL} {
+		if engineType.String() == e.String() {
+			return true
+		}
+	}
+	return false
+}
+
+// NewDatastoreEngine inits a new valid DatastoreEngine type
+func NewDatastoreEngine(engine string) (DatastoreEngine, error) {
+	dsEngine := DatastoreEngine(engine)
+	if isValid := dsEngine.IsValid(); !isValid {
+		return "", fmt.Errorf("invalid datastore engine '(%s)'", engine)
```

Comment:
We have similar code here https://github.com/openfga/openfga/blob/0f3e337dec1503fa3fc6b045f2efbb27a4defa19/cmd/run/run.go#L360 and here https://github.com/openfga/openfga/blob/8dc2ed62413e7cccc7da57da057f1f1fc17dcb8a/cmd/validatemodels/validate_models.go#L70